c-bindings-gen/src/main.rs

```diff
@@ -1280,6 +1291,16 @@ fn writeln_enum<'a, 'b, W: std::io::Write>(w: &mut W, e: &'a syn::ItemEnum, type
 	write_conv!(format!("native_into(native: native{}) -> Self", e.ident), true, false);
 	writeln!(w, "}}").unwrap();
 
+	// Implement the conversion into rust only as owned object, because even when using
+	// `to_native()` the returned structure will have a copy of the data and won't reference into
+	// the original instance. Implement for Borrow<T> instead of T so that the same trait can be
+	// used on references and owned objects since the behavior doesn't change. This could cause a
+	// performance regression when used on owned objects because the inner data instead of being
+	// moved is copied, but on the other end we gain a lot in terms of convenience.
```

Comment:
I'm confused why the trait here doesn't capture both - if you just implement the trait for both types explicitly wont it "just work"?

Reply:
I'm gonna have to try it again since this went through a few iterations: initially I did that and the compiler was complaining about conflicting implementations of the same trait, but maybe now that I've changed other stuff it will work